mbed-client-libservice/nsdynmemLIB.h

```diff
@@ -55,6 +55,12 @@ typedef struct mem_stat_t {
     uint32_t heap_alloc_fail_cnt;               /**< Counter for Heap allocation fail. */
 } mem_stat_t;
 
+
+typedef uint16_t ns_mem_block_size_t; //external interface unsigned heap block size type
```

Comment:
You can make the ` int16_t` s in the stats be ` ns_mem_heap_size_t`s - picks up that other pull request, keeps binary compatibility for current builds, and gives room for future expansion.